### DISCUSSION TITLE
static export & nice url

### DISCUSSION THREAD

**Discussion by ccrch** [vercel/next.js#20469] [Help]
This is propably a silly question to ask but im a nextjs news & figuring it out takes me a lot of time :/
I want to do just simple `next build && next export` & in a browser i have `localhost:3000/some/path/to/a/file-with-static-component` which is what i want
but after exporting I get `/some/path/to/a/file-with-static-component.html`
What needs to be done to automagically have `/some/path/to/a/file-with-static-component/` clean/pretty url?

**Comment by eviva2000** [comment]
Hi, I'm also having a similar issue, I have developed a marketing website that has multiple pages and I'm hosting it on AWS. after doing "npm run build " and deployment I get the HTML format for all the pages and the URL for accessing them is fx "mywebsite.com/download.html".
Is there any way to remove the ".html" from the url?

**Reply by cobrexus** [reply]
Check my answer https://github.com/vercel/next.js/discussions/20469#discussioncomment-242825

**Comment by cobrexus** [comment]
For a page `/path/to/component.html`, within the folder `/to/`, create a folder `/component/`, rename `component.html` to `index.html`, and move it to `/path/to/component/`.

The page `/path/to/component/index.html` is now accessible via `/path/to/component/`

---

The easiest solution is just to host the project on [Vercel](https://vercel.com), and add a `vercel.json` file to the root directory of your repository, containing the following:

```json
{
    "cleanUrls": true
}
```

[Vercel docs](https://vercel.com/docs/configuration#project/clean-urls)

**Reply by ccrch** [reply]
ye but its a bit of a hassle to do it manually after the build and paths to assets probably gonna need to be adjusted

**Reply by eviva2000** [reply]
yes, agree, especially in my case in which everything is automated. I can't change the structure after "npm run build".

**Comment by eviva2000** [comment]
I managed to get the right URL but now I'm missing all the assets. How can I change the path to assets?

**Reply by cobrexus** [reply]
I've updated my answer: https://github.com/vercel/next.js/discussions/20469#discussioncomment-242825

**Reply by eviva2000** [reply]
Thank you, Aryan!  Since this, it is not a personal project (it s my project at work), I m not allowed to change the infrastructure from aws to vercel.

**Reply by cobrexus** [reply]
@eviva2000 maybe you will find [this post on Stack Overflow](https://stackoverflow.com/q/42741839/12860895) useful?

**Comment by MarcusDelvecchio** [comment]
Having the same problem currently with a file in the `pages/` directory and only able to be routed to when `.html` is added to the end of the url route.
